quarkus-picocli/src/main/java/io/quarkus/ts/qe/command/AgeCommand.java

```diff
@@ -0,0 +1,25 @@
+package io.quarkus.ts.qe.command;
+
+import jakarta.enterprise.context.ApplicationScoped;
+
+import io.quarkus.logging.Log;
+
+import picocli.CommandLine;
+
+/**
+ * According picocli guide,
+ * Beans with @CommandLine.Command should not use proxied scopes (e.g. do not use @ApplicationScope)
+ * because Picocli will not be able to set field values in such beans.
+ * So this class creation is to test it, see testAgeCommandWithApplicationScoped.
```

Comment:
Just an opinion for future test development, no action required:

Quarkus docs basically says "if you use proxied scope, define setter". And your test is "let's test it doesn't work without setter". IMO you should test that setter helps (if you want, in addition to what doesn't work). Because setter you don't test and I am not quite interested that something doesn't work if it is documented. Users shouldn't use it even if it worked considering it requires reflection and it's not considered unsupported.

Reply:
Maybe what you want to test is that `Unmatched arguments from index 3` is printed out when you try to set it? In that case it's ok, though I still think it's less important than testing what docs advise to do.